### ISSUE TITLE
a serious problem about test.py

### ISSUE THREAD

**Issue by Qianshaowei** [open-mmlab/mmdetection#5053]
hello, I found a very serious problem. The val set and the test set are the same data set, but the test result is much smaller than the mAP during verification. The bbox that outputs the test results is also terrible.![image](https://user-images.githubusercontent.com/24608219/115898391-1638d000-a490-11eb-8574-fadaadef7312.png)![image](https://user-images.githubusercontent.com/24608219/115898644-5ef08900-a490-11eb-9da1-8f1b46bca85c.png)![image](https://user-images.githubusercontent.com/24608219/115898749-7a5b9400-a490-11eb-84eb-98127f184e44.png)

**Comment by Qianshaowei** [comment]
I found that this problem occurred when I modified the structure of the original model. So how to solve this problem.

**Comment by Qianshaowei** [comment]
More specifically, this problem occurs when I directly add some attention mechanism modules to resnet or FPN based on the faster rcnn model.

**Comment by Qianshaowei** [comment]
This problem does not exist when I delete these modules and use the base model for training and testing

**Comment by Qianshaowei** [comment]
This problem does not exist when I delete these modules and use the base model for training and testing

**Comment by Qianshaowei** [comment]
I may know the problem. I didn't register the pytorch module, so the parameters were not saved.
